mpqp/core/instruction/measurement/pauli_string.py

```diff
@@ -206,6 +212,47 @@ def simplify(self, inplace: bool = False) -> PauliString:
             self._monomials = res.monomials
         return res
 
+    def hard_simplify(self, inplace: bool = False) -> PauliString:
+        """Hard simplifies the PauliString by combining identical terms and removing
```

Comment:
The docstring of this function doesn't tell us what is the difference between ``hard_simplify`` and the already existing ``simplify``.

Is it supposed to remove the identity monomial when other terms are present?

Reply:
done